### PR TITLE
[bitnami/clickhouse] Fix volumePermissions's command syntax and security context

### PR DESCRIPTION
**Pull request by spijet**
### Description of the change

The command used in `volumePermissions` init container was specified as string instead of a list of strings. This may have resulted in errors reported by Helm:
```
Error: UPGRADE FAILED: error validating "": error validating data: ValidationError(StatefulSet.spec.template.spec.initContainers[0].command): invalid type for io.k8s.api.core.v1.Container.command: got "string", expected "array"
```

Also, this chart appears to have an unused `.Values.volumePermissions.containerSecurityContext` knob and unconditionally uses the main container's securityContext for `volumePermissions` init container instead. This breaks the startup in some cases, e.g. when we require root privileges in order to set up file permissions properly. This PR makes `volumePermissions` honor the securityContext specified for it in `values.yaml`.

### Benefits

With this PR merged, both initContainers will use the same array-based command syntax.

### Possible drawbacks

(I'm not sure, please let me know if you know of any possible problems this may introduce 😅)

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by spijet** [comment]
I also found a problem where `volumePermissions` init container didn't use its' own security context and re-used the main container's one instead. [be8734f](https://github.com/bitnami/charts/pull/16245/commits/be8734f69babdaf2ac061e9d4c0e27771178a052) fixes it.

**Comment by spijet** [comment]
Also, do we really need to mount the `config` ConfigMap to `/bitnami/clickhouse/conf/default` inside the `volumePermissions` container? It doesn't look like it's used there, and the `chown` command cannot do anything to the `conf` directory anyway, as the ConfigMap is mounted read-only.

File: bitnami/clickhouse/templates/statefulset.yaml [hunk 1]
```diff
@@ -101,28 +101,30 @@ spec:
...
+              cp /tmp/certs/* /opt/bitnami/clickhouse/certs/
+              {{- if eq ( toString ( $.Values.volumePermissions.containerSecurityContext.runAsUser )) "auto" }}
+              chown -R `id -u`:`id -G | cut -d " " -f2` /opt/bitnami/clickhouse/certs/
+              {{- else }}
+              chown -R {{ $.Values.containerSecurityContext.runAsUser }}:{{ $.Values.podSecurityContext.fsGroup }} /opt/bitnami/clickhouse/certs/
+              {{- end }}
+              chmod 600 {{ include "clickhouse.tlsCertKey" $ }}
+              {{- end }}
+          securityContext:
+            runAsUser: {{ $.Values.volumePermissions.containerSecurityContext.runAsUser }}
```
Comment (migruiz4) [inline bitnami/clickhouse/templates/statefulset.yaml:127]: Could you please render the entire `volumePermissions.containerSecurityContext` object? This way users can add extra settings if needed:
```suggestion
          securityContext: {{- include "common.tplvalues.render" (dict "value" $.Values.volumePermissions.containerSecurityContext "context" $) | nindent 12 }}
```

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Hi @spijet ,

Thank you for your contribution! Could you please take a look at my suggestion?

File: bitnami/clickhouse/templates/statefulset.yaml (hunk 1 above)
Comment (spijet) [reply bitnami/clickhouse/templates/statefulset.yaml]: @migruiz4, thank you for your suggestion!

Implemented in 527a0f84. :)

**Comment by spijet** [comment]
Hello @migruiz4!

All done! Thank you for your suggestion! :)

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Hi @spijet,

While testing this feature, the changes look great, but I found this other issue.
The clickhouse deployment will fail with volume-permissions init-containers running into crashloopback with the following error:
```
$ k logs test-clickhouse-shard1-0 -c volume-permissions
chown: changing ownership of '/bitnami/clickhouse/conf/default/00_default_overrides.xml': Read-only file system
chown: changing ownership of '/bitnami/clickhouse/conf/default/..2023_04_28_10_31_51.2718970610/00_default_overrides.xml': Read-only file system
chown: changing ownership of '/bitnami/clickhouse/conf/default/..2023_04_28_10_31_51.2718970610': Read-only file system
chown: changing ownership of '/bitnami/clickhouse/conf/default/..data': Read-only file system
chown: changing ownership of '/bitnami/clickhouse/conf/default': Read-only file system
```

This is not related to your changes, but another issue with the original logic.

Could you please confirm if you have the same issue? If so, please apply the following change in your PR by removing these lines from the volume-permission container's volumeMounts:

```
- name: config
  mountPath: /bitnami/clickhouse/conf/default
```

**Comment by spijet** [comment]
I was unsure if the `config` volume is needed in some cases, but from the looks of it there's no sense in trying to `chown`/`chmod` it, since it's a ConfigMap (i.e. always mounted as read-only). Will remove it now.

**Review by migruiz4 — APPROVED** [review]
Thank you very much for applying the suggested changes! Looks good to me!

**Comment by spijet** [comment]
Thank you very much!

**Pull request merged**